src/encoded/types/patient.py

```diff
@@ -466,51 +475,62 @@ def surgery_summary(self, request, surgery=None):
+                    elif  path_report_obj['path_source_procedure'] == "path_biopsy" or path_report_obj['path_source_procedure'] == "path_metasis":
                         non_nephrectomy_dates.append(surgery_object['date'])
+
         if len(nephrectomy_dates) > 0 :
-            nephrectomy_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
-            diagnosis_date = nephrectomy_dates[0]
-        else:
-            if len(radiation) > 0:
-                # add radiation dates
-                for radiation_record in radiation:
-                    radiation_object = request.embed(radiation_record, '@@object')
-                    non_nephrectomy_dates.append(radiation_object['start_date'])
-            if len(medication) > 0:
-                # add medication dates
-                for medication_record in medication:
-                    medication_object = request.embed(medication_record, '@@object')
-                    non_nephrectomy_dates.append(medication_object['start_date'])
-
-            if len(non_nephrectomy_dates) > 0:
+                nephrectomy_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
+                diagnosis_date = nephrectomy_dates[0]
+                surgery_dates.append(nephrectomy_dates[0])
+                diagnosis_source = "Pathology Report"
+
+        elif len(non_nephrectomy_dates) > 0:
+            non_nephrectomy_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
+            surgery_dates.append(non_nephrectomy_dates[0])
+            surgery_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
+            diagnosis_date = surgery_dates[0]
+            diagnosis_source = "Pathology Report"
+
+        elif diagnosis_date_tumor_registry is not None:
+             diagnosis_date = diagnosis_date_tumor_registry
+        elif len(medication) > 0:
+            diagnosis_source = "Medication"
+            for medication_record in medication:
+                medication_object = request.embed(medication_record, '@@object')
+                non_nephrectomy_dates.append(medication_object['start_date'])
                 non_nephrectomy_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
                 diagnosis_date = non_nephrectomy_dates[0]
+
         age_range = "Unknown"
         ageString = "Unknown"
-        follow_up_duration_range="Not available"
+        follow_up_duration_range = "Not available"
 
-        if diagnosis_date != "Not available":
+        if diagnosis_date is not "Not available":
             birth_date = datetime.strptime("1800-01-01", "%Y-%m-%d")
             end_date = datetime.strptime(diagnosis_date, "%Y-%m-%d")
             age = end_date.year - birth_date.year -  ((end_date.month, end_date.day) < (birth_date.month, birth_date.day))
             ageString = str(age)
+
             if age >= 90:
                 ageString = "90 or above"
 
-
             if age >= 80:
                 age_range = "80+"
```

Comment:
this range "80+" should be "80 - 89" if there is a range of "90 or above"

Reply:
remove the 90 or above code

Reply:
Clarrifying the reason to have ageString. The 80+ is previously what Payal wanted #50